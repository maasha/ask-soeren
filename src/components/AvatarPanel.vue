<template>
  <div data-testid="avatar-panel" class="w-1/3 ml-8 h-96">
    <div class="w-60" :class="{ 'animated-head': blablabla }">
      <img id="avatar-top" src="../assets/soeren-top-anonymous.jpg" alt="avatar top" />
    </div>
    <div class="w-60" :class="{ 'animated-jaw': blablabla }">
      <img id="avatar-bottom" src="../assets/soeren-bottom.jpg" alt="avatar bottom" />
    </div>
  </div>
</template>

<script lang="ts">
import { defineComponent, PropType } from 'vue'

/**
 * Panel to display the animated SÃ¸ren avatar.
 */
export default defineComponent({
  props: {
    /**
     * Flag indicating that the avatar must be talking.
     */
    blablabla: {
      type: Boolean as PropType<boolean>,
      required: true,
    },
  },
})
</script>

<style scoped>
@keyframes moveJaw {
  0% {
    padding-top: 0;
  }
  50% {
    padding-top: 20px;
  }
  100% {
    padding-top: 0;
  }
}

@keyframes rotateHead {
  0% {
    transform: rotate(0);
  }
  25% {
    transform: rotate(-1deg);
  }
  50% {
    transform: rotate(0);
  }
  75% {
    transform: rotate(1deg);
  }
  100% {
    transform: rotate(0deg);
  }
}

.animated-head {
  animation: rotateHead 1s infinite;
}

.animated-jaw {
  animation: moveJaw 0.2s infinite;
}
</style>
