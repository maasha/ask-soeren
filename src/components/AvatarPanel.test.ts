import { render } from '@testing-library/vue'

import AvatarPanel from './AvatarPanel.vue'

describe('AvatarPanel', async () => {
  test('should match snapshot', () => {
    const renderResult = render(AvatarPanel)
    expect(renderResult).toMatchSnapshot()
  })
})
