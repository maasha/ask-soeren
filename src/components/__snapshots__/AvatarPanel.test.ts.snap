// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AvatarPanel > should match snapshot 1`] = `
{
  "baseElement": <body>
    <div>
      <div
        class="w-1/3 ml-8 h-96"
        data-testid="avatar-panel"
        data-v-4e7fa8f4=""
      >
        <div
          class="w-60"
          data-v-4e7fa8f4=""
        >
          <img
            alt="avatar top"
            data-v-4e7fa8f4=""
            id="avatar-top"
            src="/src/assets/soeren-top-anonymous.jpg"
          />
        </div>
        <div
          class="w-60"
          data-v-4e7fa8f4=""
        >
          <img
            alt="avatar bottom"
            data-v-4e7fa8f4=""
            id="avatar-bottom"
            src="/src/assets/soeren-bottom.jpg"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-1/3 ml-8 h-96"
      data-testid="avatar-panel"
      data-v-4e7fa8f4=""
    >
      <div
        class="w-60"
        data-v-4e7fa8f4=""
      >
        <img
          alt="avatar top"
          data-v-4e7fa8f4=""
          id="avatar-top"
          src="/src/assets/soeren-top-anonymous.jpg"
        />
      </div>
      <div
        class="w-60"
        data-v-4e7fa8f4=""
      >
        <img
          alt="avatar bottom"
          data-v-4e7fa8f4=""
          id="avatar-bottom"
          src="/src/assets/soeren-bottom.jpg"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "emitted": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "html": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
